packages/backend/src/models/UserModel.ts

```diff
@@ -98,27 +271,92 @@ export class UserModel {
             .select<DbOrganizationUser[]>([
                 'users.user_uuid',
                 'users.first_name',
                 'users.last_name',
                 'emails.email',
-            ])
-            .where('organization_uuid', organizationUuid);
+            ]);
+    }
+
+    async createInitialUser({
+        firstName,
+        lastName,
+        organizationName,
+        email,
+        password,
+        isMarketingOptedIn,
+        isTrackingAnonymized,
+    }: CreateInitialUserArgs): Promise<LightdashUser> {
+        const user = await this.database.transaction(async (trx) => {
+            try {
+                const newOrg = await createOrganization(trx, {
+                    organization_name: organizationName.trim(),
+                });
+                const [newUser] = await trx<DbUser>('users')
+                    .insert<DbUserIn>({
+                        first_name: firstName.trim(),
+                        last_name: lastName.trim(),
+                        is_marketing_opted_in: isMarketingOptedIn,
+                        is_tracking_anonymized: isTrackingAnonymized,
+                    })
+                    .returning('*');
+                await createEmail(trx, {
+                    user_id: newUser.user_id,
+                    email,
+                    is_primary: true,
+                });
+                await createPasswordLogin(trx, {
+                    user_id: newUser.user_id,
+                    password_hash: await bcrypt.hash(
+                        password,
+                        await bcrypt.genSalt(),
+                    ),
+                });
+                await createOrganizationMembership(trx, {
+                    organization_id: newOrg.organization_id,
+                    user_id: newUser.user_id,
+                });
+
+                return newUser;
+            } catch (e) {
+                await trx.rollback(e);
+                throw e;
+            }
+        });
+        return this.getUserDetailsByUuid(user.user_uuid);
+    }
+
+    async findSessionUserByUUID(userUuid: string): Promise<SessionUser> {
+        const [user] = await userDetailsQueryBuilder(this.database)
+            .where('user_uuid', userUuid)
+            .select('*');
+        if (user === undefined) {
+            throw new NotFoundError(`Cannot find user with uuid ${userUuid}`);
+        }
+        return {
+            userId: user.user_id,
+            ...mapDbUserDetailsToLightdashUser(user),
+        };
+    }
+
+    async findSessionUserByPrimaryEmail(email: string): Promise<SessionUser> {
+        const [user] = await userDetailsQueryBuilder(this.database)
+            .where('email', email)
+            .select('*');
+        if (user === undefined) {
+            throw new NotFoundError(`Cannot find user with uuid ${email}`);
+        }
+        return {
+            userId: user.user_id,
+            ...mapDbUserDetailsToLightdashUser(user),
+        };
+    }
```

Comment:
These are a little frustrating and exist only so we can attach the `userId` to the Session. How annoying does it become if we only attach the `userUuid` to the session?

Reply:
Yeah, that would be nice.